bluepysnap/_plotting.py

```diff
@@ -0,0 +1,372 @@
+            and plt.figure() are used).
+
+    Returns :
+        (matplotlib.animation.FuncAnimation, matplotlib.Axis): the matplotlib animation object and
+            the corresponding axis.
+
+    Notes:
+        From scripts:
+        >>> import matplotlib.pyplot as plt
+        >>> from bluepysnap import Simulation
+        >>> report = Simulation("config.json").spikes["my_population"]
+        >>> anim, ax = report.firing_animation()
+        >>> plt.show()
+        >>> # to save the animation : do not plt.show() and just anim.save('my_movie.mp4')
+
+        From notebooks:
+        >>> from IPython.display import HTML
+        >>> from bluepysnap import Simulation
+        >>> report = Simulation("config.json").spikes["my_population"]
+        >>> anim, ax = report.firing_animation()
+        >>> HTML(anim.to_html5_video())
+    """
+    plt = _get_pyplot()
+    from matplotlib.animation import FuncAnimation
+
+    def _check_axis(axis):
+        """Verifies axes values."""
+        axes = {Node.X, Node.Y, Node.Z}
+        if axis not in axes:
+            raise BluepySnapError('{} is not a valid axis'.format(axis))
+
+    _check_axis(x_axis)
+    _check_axis(y_axis)
+
+    spike_report = filtered_report.spike_report
+    population_names = filtered_report.spike_report.population_names
+    report = filtered_report.report
+
+    data = pd.DataFrame(index=report.index, columns=[x_axis, y_axis], dtype=np.float32)
+
+    for population in population_names:
+        spikes = spike_report[population]
+        pop_mask = report["population"] == population
+
+        ids = report.loc[pop_mask, "ids"].to_numpy()
+        try:
+            values = spikes.nodes.get(properties=[x_axis, y_axis]).loc[ids].to_numpy()
+            data.loc[pop_mask, [x_axis, y_axis]] = values
+        except BluepySnapError:
+            continue
+
+    data = data[data.notna()]
+
+    if ax is None:
+        fig = plt.figure()
+        ax = plt.gca()
+        ax.set_title('time = {}ms'.format(np.min(data.index)))
+        x_limits = [data[x_axis].min(), data[x_axis].max()]
+        y_limits = [data[y_axis].min(), data[y_axis].max()]
+        ax.set_xlim(*x_limits)
+        ax.set_ylim(*y_limits)
+        ax.set_xlabel('{} $\mu$m'.format(x_axis))  # noqa
+        ax.set_ylabel('{} $\mu$m'.format(y_axis))  # noqa
+
+    else:
+        fig = ax.figure
+
+    dots = ax.plot([], [], '.k')
+
+    def update_animation(frame):
+        """Update the animation plots and axes."""
+        ax.set_title('time = ' + str(frame * dt) + ' ms')
+        mask = (data.index >= frame * dt) & (data.index <= (frame + 1) * dt)
+        positions = data.loc[mask, [x_axis, y_axis]].values
+        x = positions[:, 0]
+        y = positions[:, 1]
+        dots[0].set_data(x, y)
+        return dots
+
+    frames = list(range(int(data.index[0] / dt), int(data.index[-1] / dt)))
+    anim = FuncAnimation(fig, update_animation, frames=frames)
```

Comment:
Because matplotlib is utterly bad and works with some random global contexts.
You need to return the ax AND the anim if you want the animation to run. If you return the ax only you gonna have only the first frame. Don't ask why...